arch/HPPA/HPPAInstPrinter.c

```diff
@@ -0,0 +1,532 @@
+	{ HPPA_INS_FSQRT, "RW", false },
+	{ HPPA_INS_FABS, "RW", false },
+	{ HPPA_INS_FREM, "", false },
+	{ HPPA_INS_FRND, "RW", false },
+	{ HPPA_INS_FCPY, "RW", false },
+	{ HPPA_INS_FCNVFF, "RW", false },
+	{ HPPA_INS_FCNVXF, "RW", false },
+	{ HPPA_INS_FCNVFX, "RW", false },
+	{ HPPA_INS_FCNVFXT, "RW", false },
+	{ HPPA_INS_FMPYFADD, "RRRW", false },
+	{ HPPA_INS_FMPYNFADD, "RRRW", false },
+	{ HPPA_INS_FNEG, "RW", false },
+	{ HPPA_INS_FNEGABS, "RW", false },
+	{ HPPA_INS_FCNV, "RW", false },
+	{ HPPA_INS_FCMP, "RR", false },
+	{ HPPA_INS_FCMP, "RRi", true },
+	{ HPPA_INS_XMPYU, "RRW", false },
+	{ HPPA_INS_FMPYADD, "RRWRw", false },
+	{ HPPA_INS_FMPYSUB, "RRWRw", false },
+	{ HPPA_INS_FTEST, "", false },
+	{ HPPA_INS_FTEST, "i", true },
+	{ HPPA_INS_FID, "", false },
+
+	{ HPPA_INS_PMDIS, "", false },
+	{ HPPA_INS_PMENB, "", false },
+
+	{ HPPA_INS_SPOP0, "", false },
+	{ HPPA_INS_SPOP1, "W", false },
+	{ HPPA_INS_SPOP2, "R", false },
+	{ HPPA_INS_SPOP3, "RR", false },
+	{ HPPA_INS_COPR, "", false },
+	{ HPPA_INS_CLDW, "x(Rb)W", false },
+	{ HPPA_INS_CLDD, "o(Rb)W", false },
+	{ HPPA_INS_CSTW, "Rx(Rb)", false },
+	{ HPPA_INS_CSTD, "Rx(Rb)", false },
+	{ HPPA_INS_CLDWX, "r(Rb)W", false },
+	{ HPPA_INS_CLDDX, "r(Rb)W", false },
+	{ HPPA_INS_CSTWX, "Rr(Rb)", false },
+	{ HPPA_INS_CSTDX, "Rr(Rb)", false },
+	{ HPPA_INS_CLDWS, "o(Rb)W", false },
+	{ HPPA_INS_CLDDS, "o(Rb)W", false },
+	{ HPPA_INS_CSTWS, "Ro(Rb)", false },
+	{ HPPA_INS_CSTDS, "Ro(Rb)", false },
+
+	{ HPPA_INS_CALL, "", false },
+	{ HPPA_INS_RET, "", false },
+};
+
+static void print_operand(MCInst *MI, struct SStream *O, const cs_hppa_op *op)
+{
+	switch (op->type) {
+	case HPPA_OP_INVALID:
+		SStream_concat(O, "invalid");
+		break;
+	case HPPA_OP_REG:
+		SStream_concat(O, HPPA_reg_name((csh)MI->csh, op->reg));
+		break;
+	case HPPA_OP_IMM:		
+		printInt32(O, op->imm);
+		break;
+    case HPPA_OP_DISP:
+		printInt32(O, op->imm);
+		break;
+	case HPPA_OP_IDX_REG:
+		SStream_concat(O, HPPA_reg_name((csh)MI->csh, op->reg));
+		break;
+	case HPPA_OP_MEM:
+		SStream_concat(O, "(");
+		if (op->mem.space != HPPA_OP_INVALID && op->mem.space != HPPA_REG_SR0) {
+			SStream_concat(O, HPPA_reg_name((csh)MI->csh, op->mem.space));
+			SStream_concat(O, ",");
+		}
+		SStream_concat(O, HPPA_reg_name((csh)MI->csh, op->mem.base));
+		SStream_concat(O, ")");
+		break;
+	case HPPA_OP_TARGET:
+		printUInt64(O, MI->address + op->imm);
+		break;
+	}
+}
+
+#define NUMFMTS ((sizeof pa_formats)/(sizeof pa_formats[0]))
+
+static void fill_operands(MCInst *MI, cs_hppa *hppa)
+{
+	hppa->op_count = 0;
+	unsigned mc_op_count = MCInst_getNumOperands(MI);
+	if (mc_op_count == 0)
+		return;
+
+	hppa_ext *hppa_ext = &MI->hppa_ext;
+	uint32_t opcode = MCInst_getOpcode(MI);
+
+    for (int i = 0; i < NUMFMTS; ++i) {
+		const struct pa_insn_fmt *pa_fmt = &pa_formats[i];
+		if (opcode == pa_fmt->insn_id && hppa_ext->is_alternative == pa_fmt->is_alternative) {
+			char *fmt = (char *)pa_fmt->format;
```

Comment:
```suggestion
			const char *fmt = pa_fmt->format;
```

Any reason this is not `const`?

Reply:
Yes. We use `fmt` to iterate over `pa_fmt->format`  to select the kind of operand in instruction

Reply:
Ah yes. But does it through any warnings for you? I think for me it was fine. You should be able to iterate also over an `const` array of structs, since no value is changed.